field_RANKSUFF_module.fypp

```diff
@@ -485,6 +502,16 @@ CONTAINS
 
   END SUBROUTINE ${ftn}$_GET_HOST_DATA_RDWR
 
+  SUBROUTINE ${ftn}$_WRAPPER_GET_HOST_DATA_RDWR (SELF, PPTR, QUEUE)
+    CLASS(${ftn}$_WRAPPER) :: SELF
+    ${ft.type}$, POINTER,         INTENT(INOUT) :: PPTR(${ft.shape}$)
+    INTEGER(KIND=JPIM), OPTIONAL, INTENT(IN)    :: QUEUE
+
+    CALL SELF%GET_HOST_DATA (IOR (NRD, NWR), PPTR, QUEUE)
+    SELF%INITIALIZED = .TRUE.
```

Comment:
`OWNER` fields already have a mechanism to prevent HtoD copies by setting the status to `UNINITIALIZED`. Perhaps I've missed something, but could we not extend that same mechanism to `WRAPPER` fields? This would be cleaner imo rather than having two different mechanisms of checking whether a field is unitialized. This would also allow `GET_DEVICE_DATA` to remain unchanged, rather than duplicating the same code in `WRAPPER_GET_DEVICE_DATA` and `OWNER_GET_DEVICE_DATA`.